core/src/main/java/org/ehcache/core/config/DefaultConfiguration.java

```diff
@@ -77,7 +78,7 @@ public DefaultConfiguration(ClassLoader classLoader, ServiceCreationConfiguratio
   public DefaultConfiguration(Map<String, CacheConfiguration<?, ?>> caches, ClassLoader classLoader, ServiceCreationConfiguration<?>... services) {
     this.services = unmodifiableCollection(Arrays.asList(services));
     this.caches = new ConcurrentHashMap<String, CacheConfiguration<?, ?>>(caches);
-    this.classLoader = classLoader;
+    this.classLoader = classLoader == null ? ClassLoading.getDefaultClassLoader() : classLoader;
```

Comment:
Oh sorry! Misunderstanding ;-) By null-check I meant:

```
if(classloader == null) throw new NPE()
```

The idea behind that is that Configuration.getClassloader() contract says classloader should not be null; So we can make sure of that in the copy constructor.

Also, the idea of this null check is to have a real copy constructor: if we init the classloader to `ClassLoading.getDefaultClassLoader()` if `Configuration.getClassloader()` is null, then it means that the classloader is not "copied" from the Configuration object and `DefaultConfiguration.getClassloader()` would be different than `Configuration.getClassloader()`

So perhaps it would be better to keep this where it is now:

```
this.classLoader = classLoader == null ? ClassLoading.getDefaultClassLoader() : classLoader
```

... and just add a null check in the copy ctor ?

```
if(classloader == null) throw new NPE()
```